src/spikeinterface/extractors/cellexplorersortingextractor.py

```diff
@@ -1,136 +1,163 @@
-            ), f"No {sorting_id}.sessionInfo.mat file found in the folder!" 
-
-            try:
-                session_info_mat = scipy.io.loadmat(file_name=str(session_info_matfile_path))
-                self.read_session_info_with_scipy = True
-            except NotImplementedError:
-                session_info_mat = hdf5storage.loadmat(file_name=str(session_info_matfile_path))
-                self.read_session_info_with_scipy = False
-            
-            assert session_info_mat["sessionInfo"]["rates"][0][0]["wideband"], (
-                "The sesssionInfo.mat file must contain "
-                "a 'sessionInfo' struct with field 'rates' containing field 'wideband' to extract the sampling frequency!"
-            )
-            if self.read_session_info_with_scipy:
-                sampling_frequency = float(
-                    session_info_mat["sessionInfo"]["rates"][0][0]["wideband"][0][0][0][0]
-                )  # careful not to confuse it with the lfpsamplingrate; reported in units Hz
-            else:
-                sampling_frequency = float(
-                    session_info_mat["sessionInfo"]["rates"][0][0]["wideband"][0][0]
-                )  # careful not to confuse it with the lfpsamplingrate; reported in units Hz
+            # First try is for the new format of spikes.cellinfo.mat files where sampling rate is included
+            sampling_frequency = spikes_mat["spikes"].get("sr", None)
+            sampling_frequency = sampling_frequency[()] if sampling_frequency is not None else None 
+        
+        if sampling_frequency is None:
+            sampling_frequency = self._retrieve_sampling_frequency_from_session_info()
+        
+        sampling_frequency = float(sampling_frequency)
+        
+        unit_ids_available = "UID" in spikes_mat["spikes"].keys()
+        assert unit_ids_available, "The spikes.cellinfo.mat file must contain field 'UID'!"
 
-        try:
-            spikes_mat = scipy.io.loadmat(file_name=str(spikes_matfile_path))
-            self.read_spikes_info_with_scipy = True
-        except NotImplementedError: 
-            spikes_mat = hdf5storage.loadmat(file_name=str(spikes_matfile_path))
-            self.read_spikes_info_with_scipy = False
+        spike_times_available = "times" in spikes_mat["spikes"].keys()
+        assert spike_times_available, "The spikes.cellinfo.mat file must contain field 'times'!"
 
-        assert np.all(
-            np.isin(["UID", "times"], spikes_mat["spikes"].dtype.names)
-        ), "The spikes.cellinfo.mat file must contain a 'spikes' struct with fields 'UID' and 'times'!"
+        unit_ids = spikes_mat["spikes"]["UID"]
+        spike_times = spikes_mat["spikes"]["times"]
+        
+        if read_as_hdf5:
+            unit_ids = unit_ids[:].squeeze().astype("int")
+            spike_references = spike_times[:]  # These are HDF5 references
+            spike_times = [spikes_mat[reference[0]][()].squeeze() for reference in spike_references]
```

Comment:
Oh I see, conditional reshaping is done after the fact... But the availability of outer keys does not change from scipy vs h5py